flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointFailureManager.java

```diff
@@ -59,6 +61,23 @@ public CheckpointFailureManager(int tolerableCpFailureNumber, FailJobCallback fa
 	 *                      latest generated checkpoint id as a special flag.
 	 */
 	public void handleCheckpointException(CheckpointException exception, long checkpointId) {
+		handleCheckpointException(exception, checkpointId, null);
+	}
+
+	/**
+	 * Handle checkpoint exception with a handler callback.
+	 *
+	 * @param exception the checkpoint exception.
+	 * @param checkpointId the failed checkpoint id used to count the continuous failure number based on
+	 *                     checkpoint id sequence. In trigger phase, we may not get the checkpoint id when the failure
+	 *                     happens before the checkpoint id generation. In this case, it will be specified a negative
+	 *                      latest generated checkpoint id as a special flag.
+	 * @param executionAttemptID the execution attempt id, as a safe guard.
+	 */
+	public void handleCheckpointException(
+		CheckpointException exception,
```

Comment:
nit: formatting. Usually we tend to double intend (with two tabs) the parameters list, to avoid first line of method's body "mangling" with the parameters.